### PR TITLE
Fix __assert.h so it can be included in C++ code

### PR DESCRIPTION
**Pull request by lowlander**
Add extern "C" to __assert.h so it can be included into C++ code.

### CODE REVIEW

File: include/sys/__assert.h [hunk 1]
```diff
@@ -63,6 +63,10 @@
 
 #include <stdbool.h>
 
+#ifdef __cplusplus
```
Comment (pabigot) [inline include/sys/__assert.h]: This has to be moved into the `#if __ASSERT_ON` block and be positioned after `#include <sys/printk.h>`.

File: include/sys/__assert.h [hunk 2]
```diff
@@ -126,4 +130,8 @@ void assert_post_action(const char *file, unsigned int line);
 #define __ASSERT_NO_MSG(test) { }
 #endif
 
+#ifdef __cplusplus
```
Comment (pabigot) [inline include/sys/__assert.h]: This will have to move to be at the same level as the moved `extern "C"`.

**Review by pabigot — CHANGES_REQUESTED** [review]
The `extern "C"` block must appear after all `#include` directives.

File: include/sys/__assert.h (hunk 2 above)
Comment (lowlander) [reply include/sys/__assert.h]: @pabigot  better ?

**Review by pabigot — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**